R/doubletFinder.R

```diff
@@ -5,59 +5,69 @@
 #' artificial nearest neighbors is defined for each real cell. Finally, real cells are rank-
 #' ordered and predicted doublets are defined via thresholding based on the expected number
-#' of doublets.
+#' of doublets. This function has methods for both Seurat and loom objects. To use the method for loom,
+#' please install loomR (from GitHub, not on CRAN yet) and the loom branch of Seurat. See
+#' https://satijalab.org/seurat/mca_loom.html for more instructions.
 #'
-#' @param seu A fully-processed Seurat object (i.e. after normalization, variable gene definition,
+#' @param obj A fully-processed Seurat or loom object (i.e. after normalization, variable gene definition,
 #' scaling, PCA, and tSNE).
+#' @rdname doubletFinder
+#' @export doubletFinder
+#'
+doubletFinder <- function(obj, ...) {
+  UseMethod(generic = "doubletFinder", object = obj)
+}
+
 #' @param expected.doublets The number of doublets expected to be present in the original data.
 #' This value can best be estimated from cell loading densities into the 10X/Drop-Seq device.
-#' @param porportion.artificial The proportion (from 0-1) of the merged real-artificial dataset
+#' @param proportion.artificial The proportion (from 0-1) of the merged real-artificial dataset
 #' that is artificial. In other words, this argument defines the total number of artificial doublets.
-#' Default is set to 25%, based on optimization on PBMCs (see McGinnis, Murrow and Gartner 2018, BioRxiv).
+#' Default is set to 25\%, based on optimization on PBMCs (see McGinnis, Murrow and Gartner 2018, BioRxiv).
 #' @param proportion.NN The proportion (from 0-1) of the merged real-artificial dataset used to define
-#' each cell's neighborhood in PC space. Default set to 1%, based on optimization on PBMCs (see McGinnis,
+#' each cell's neighborhood in PC space. Default set to 1\%, based on optimization on PBMCs (see McGinnis,
 #' Murrow and Gartner 2018, BioRxiv).
 #' @return An updated Seurat object with metadata for pANN values and doublet predictions.
-#' @export
+#' @describeIn doubletFinder Doublet detection in single-cell RNA sequencing data
+#' @export doubletFinder.seurat
+#' @method doubletFinder seurat
 #' @examples
-#' seu <- doubletFinder(seu, expected.doublets = 1000, proportion.artificial = 0.25, proportion.NN = 0.01)
-
-
-doubletFinder <- function(seu, expected.doublets = 0, proportion.artificial = 0.25, proportion.NN = 0.01) {
+#' seu <- doubletFinder(obj, expected.doublets = 1000, proportion.artificial = 0.25, proportion.NN = 0.01)
+#'
+doubletFinder.seurat <- function(obj, expected.doublets = 0, proportion.artificial = 0.25, proportion.NN = 0.01) {
 
   if (expected.doublets == 0) {  stop("Need to set number of expected doublets...")  }
 
   ## Step 1: Generate artificial doublets from Seurat object input
   print("Creating artificial doublets...")
-  real.cells <- colnames(seu@data)
-  data <- seu@raw.data[ , real.cells]
+  real.cells <- colnames(obj@data)
```

Comment:
Unless there is a compelling reason to switch, I'd like to keep 'seu' over 'obj'

Reply:
The reason is that if we use loom objects, the class is loom rather than Seurat, so using seu may be confusing to users. Anyway, you can change it back.